example/e2e/asyncstorage.spec.js

```diff
@@ -139,4 +152,84 @@ describe('Async Storage', () => {
       expect(storyText).toHaveText(newStory);
     });
   });
+
+  describe('get / set / clear item delegate test', () => {
```

Comment:
I feel like we should rather focus on testing if delegate is set/unset, otherwise we'd just repeated test that we already have.

Reply:
Did you have anything special in mind? This change should be transparent from JS perspective. Would it make more sense to make all the delegate tests throw the keys/values it received, perhaps? That way we know for sure the delegate is handling the requests?

Reply:
Sounds like a good idea - that way, we can be sure that the delegate is responding back, not default implementation.

Reply:
Unfortunately, I couldn't get this to work reliably at all. Instead, I've rewritten the tests to expect different values coming from the delegate. For the get/set/clear tests, the delegate will always return the expected value + 1000000. For the merge test, "from delegate" string was appended to some fields. Please take a look 😄